hoss/coordinate_utilities.py

```diff
@@ -0,0 +1,232 @@
+        if varinfo.get_variable(coordinate).is_longitude()
+    ]
+
+    return latitude_coordinate_variables, longitude_coordinate_variables
+
+
+def get_row_col_sizes_from_coordinate_datasets(
+    lat_arr: ndarray,
+    lon_arr: ndarray,
+) -> Tuple[int, int]:
+    """
+    This method returns the row and column sizes of the coordinate datasets
+
+    """
+    if lat_arr.ndim > 1 and lon_arr.shape == lat_arr.shape:
+        col_size = lat_arr.shape[1]
+        row_size = lat_arr.shape[0]
+    elif (
+        lat_arr.ndim == 1
+        and lon_arr.ndim == 1
+        and lat_arr.size > 0
+        and lon_arr.size > 0
+    ):
+        col_size = lon_arr.size
+        row_size = lat_arr.size
+    else:
+        raise IrregularCoordinateVariables(lon_arr.shape, lat_arr.shape)
+    return row_size, col_size
+
+
+def get_lat_lon_arrays(
+    prefetch_dataset: Dataset,
+    latitude_coordinate: VariableFromDmr,
+    longitude_coordinate: VariableFromDmr,
+) -> Tuple[ndarray, ndarray]:
+    """
+    This method is used to return the lat lon arrays from a 2D
+    coordinate dataset.
+    """
+    try:
+        lat_arr = prefetch_dataset[latitude_coordinate.full_name_path][:]
+    except Exception as exception:
+        raise MissingCoordinateVariable(
+            latitude_coordinate.full_name_path
+        ) from exception
+
+    try:
+        lon_arr = prefetch_dataset[longitude_coordinate.full_name_path][:]
+    except Exception as exception:
+        raise MissingCoordinateVariable(
+            longitude_coordinate.full_name_path
+        ) from exception
+
+    return lat_arr, lon_arr
+
+
+def get_dimension_scale_from_dimvalues(
+    dim_values: ndarray, dim_indices: ndarray, dim_size: float
+) -> ndarray:
+    """
+    return a full dimension scale based on the 2 projected points and
+    grid size
+    """
+    dim_resolution = 0.0
+    if (dim_indices[1] != dim_indices[0]) and (dim_values[1] != dim_values[0]):
+        dim_resolution = (dim_values[1] - dim_values[0]) / (
+            dim_indices[1] - dim_indices[0]
+        )
+    if dim_resolution == 0.0:
+        raise CannotComputeDimensionResolution(dim_values[0], dim_indices[0])
+
+    # create the dim scale
+    dim_asc = dim_values[1] > dim_values[0]
+
+    if dim_asc:
+        dim_min = dim_values[0] + (dim_resolution * dim_indices[0])
+        dim_max = dim_values[0] + (dim_resolution * (dim_size - dim_indices[0] - 1))
+        dim_data = np.linspace(dim_min, dim_max, dim_size)
+    else:
+        dim_max = dim_values[0] + (-dim_resolution * dim_indices[0])
+        dim_min = dim_values[0] - (-dim_resolution * (dim_size - dim_indices[0] - 1))
+        dim_data = np.linspace(dim_max, dim_min, dim_size)
+
+    return dim_data
+
+
+def get_valid_indices(
+    coordinate_row_col: ndarray, coordinate_fill: float, coordinate_name: str
+) -> ndarray:
+    """
+    Returns indices of a valid array without fill values
+    """
+
+    if coordinate_fill:
```

Comment:
right. if we have a fill value - we use that to check for validity of the data and if that is not available we check for the geo extent range. I guess we can check for it even if the fill value is provided - in case the coordinate data itself is bad data.

Reply:
I would check the coordinates regardless, and let the fill-value be outside that range.  It simplifies the code and those checks need to happen.

Reply:
updated - 51b110c7294791d63551dc14d8ada49573eafeb0